lib/prometheus-parser/src/lib.rs

```diff
@@ -1,12 +1,34 @@
+use indexmap::IndexMap;
```

Comment:
Is there a particular benefit we need from this data structure over `HashMap` or `BTreeMap`?

Reply:
`IndexMap` preserves iteration order.

Reply:
I am aware, but is that a property we explicitly make use of? Alternatively, `BTreeMap` sorts by keys and could be used when an order is implemented on `GroupKey` (`timestamp` looks like a good candidate).

Reply:
I'm not sure, haven't dug much into this code, but I know we rely on insertion order in topology. I know we use `BTreeMap` elsewhere for its implicit key sorting-- in `top`, for example. @bruceg will know.

Reply:
We want to preserve the order of metrics as they arrived within a batch. Further, some complex metrics like histograms have some dependencies on that ordering being correct.